homeassistant/components/media_player/frontier_silicon.py

```diff
@@ -0,0 +1,254 @@
+        """Initialize the Frontier Silicon API device."""
+        self._device_url = device_url
+        self._password = password
+        self._state = STATE_UNKNOWN
+
+        self._name = None
+        self._title = None
+        self._artist = None
+        self._album_name = None
+        self._mute = None
+        self._source = None
+        self._source_list = None
+        self._media_image_url = None
+
+    # Properties
+    @property
+    def fs_device(self):
+        """
+        Create a fresh fsapi session.
+
+        A new session is created for each request in case someone else
+        connected to the device in between the updates and invalidated the
+        existing session (i.e UNDOK).
+        """
+        from fsapi import FSAPI
+
+        return FSAPI(self._device_url, self._password)
+
+    @property
+    def should_poll(self):
+        """Device should be polled."""
+        return True
+
+    @property
+    def name(self):
+        """Return the device name."""
+        return self._name
+
+    @property
+    def media_title(self):
+        """Title of current playing media."""
+        return self._title
+
+    @property
+    def media_artist(self):
+        """Artist of current playing media, music track only."""
+        return self._artist
+
+    @property
+    def media_album_name(self):
+        """Album name of current playing media, music track only."""
+        return self._album_name
+
+    @property
+    def media_content_type(self):
+        """Content type of current playing media."""
+        return MEDIA_TYPE_MUSIC
+
+    @property
+    def supported_features(self):
+        """Flag of media commands that are supported."""
+        return SUPPORT_FRONTIER_SILICON
+
+    @property
+    def state(self):
+        """Return the state of the player."""
+        return self._state
+
+    # source
+    @property
+    def source_list(self):
+        """List of available input sources."""
+        return self._source_list
+
+    @property
+    def source(self):
+        """Name of the current input source."""
+        return self._source
+
+    @property
+    def media_image_url(self):
+        """Image url of current playing media."""
+        return self._media_image_url
+
+    def update(self):
+        """Get the latest date and update device state."""
+        fs_device = self.fs_device
+
+        if not self._name:
+            self._name = fs_device.friendly_name
+
+        if not self._source_list:
+            self._source_list = fs_device.mode_list
+
+        status = fs_device.play_status
+        self._state = {
+            'playing': STATE_PLAYING,
+            'paused': STATE_PAUSED,
+            'stopped': STATE_OFF,
+            'unknown': STATE_UNKNOWN,
+            None: STATE_OFF,
+        }.get(status, STATE_UNKNOWN)
+
+        info_name = fs_device.play_info_name
+        info_text = fs_device.play_info_text
+
+        self._title = '- '.join(filter(None, [info_name, info_text]))
```

Comment:
If the dash will be the separator we should have a space on both sides, so `' - '.join(`